tests/compiler/test_plugins.py

```diff
@@ -435,6 +442,75 @@ def test_native_types(snippetcompiler: "SnippetCompilationTest") -> None:
 a = plugin_native_types::many_arguments(["a","c","b"], 1)
 
 none = plugin_native_types::as_none("a")
+
+# Union types (input)
+plugin_native_types::union_single_type(value="test")     # type value: Union[str]
+plugin_native_types::union_multiple_types(value="test")  # type value: Union[int, str]
+plugin_native_types::union_multiple_types(value=123)     # type value: Union[int, str]
+for val in ["test", 123, null]:
+    plugin_native_types::union_optional_1(value=val)     # type value: Union[None, int, str]
+    plugin_native_types::union_optional_2(value=val)     # type value: Optional[Union[int, str]]
+    plugin_native_types::union_optional_3(value=val)     # type value: Union[int, str] | None
+    plugin_native_types::union_optional_4(value=val)     # type value: None | Union[int, str]
+end
+
+# Union types (return value)
+plugin_native_types::union_return_single_type(value="test")     # type return value: Union[str]
+plugin_native_types::union_return_multiple_types(value="test")  # type return value: Union[str, int]
+plugin_native_types::union_return_multiple_types(value=123)     # type return value: Union[str, int]
+for val in ["test", 123, null]:
+    plugin_native_types::union_return_optional_1(value=val)     # type return value: Union[None, int, str]
+    plugin_native_types::union_return_optional_2(value=val)     # type return value: Optional[Union[int, str]]
+    plugin_native_types::union_return_optional_3(value=val)     # type return value: Union[int, str] | None
+    plugin_native_types::union_return_optional_4(value=val)     # type return value: None | Union[int, str]
+end
         """
     )
     compiler.do_compile()
+
```

Comment:
These different scenario's result in different types of exceptions (`RuntimeException`, `WrappingRuntimeException` and `TypingException`). Is that desired?

Reply:
`WrappingRuntimeException` is expected for sure. I already have another comment about the runtime vs typing exception. I'm a bit surprised though that they aren't all wrapped.